request_logging/middleware.py

```diff
@@ -180,29 +191,33 @@ def _log_request_body(self, request, logging_context):
                 self.logger.log(self.log_level, self._chunked_to_max(request.body), logging_context)
 
     def process_response(self, request, response):
-        resp_log = "{} {} - {}".format(request.method, request.get_full_path(), response.status_code)
-        skip_logging_because = self._should_log_route(request)
-        if skip_logging_because:
-            self.logger.log_error(logging.INFO, resp_log, {'args': {}, 'kwargs': { 'extra' :  { 'no_logging': skip_logging_because } }})
-            return response
-        logging_context = self._get_logging_context(request, response)
-
-        if response.status_code in range(400, 500):
-            if self.http_4xx_log_level == DEFAULT_HTTP_4XX_LOG_LEVEL:
-                # default, log as per 5xx
+        if response:
+            resp_log = "{} {} - {}".format(request.method, request.get_full_path(), response.status_code)
+            skip_logging_because = self._should_log_route(request)
+            if skip_logging_because:
+                self.logger.log_error(logging.INFO, resp_log, {'args': {}, 'kwargs': { 'extra' :  { 'no_logging': skip_logging_because } }})
+                return response
+            logging_context = self._get_logging_context(request, response)
+
+            if response.status_code in range(400, 500):
```

Comment:
This method has gotten ugly over time and needs to be refactored into smaller functions if you see any easy improvements. In addition, it would be good to extract the `range(400, 500` and `range(500, 600)` into constants at the module level rather than calculating them each time.

Reply:
I didnt touch those. Might be the formatting.

Reply:
Fair enough, probably bad whitespace hygiene by one of us in the past. Do you mind extracting those two ranges and add a test and we can call it a day?